ngraph/core/src/op/squeeze.cpp

```diff
@@ -149,28 +149,40 @@ shared_ptr<Node> op::Squeeze::clone_with_new_inputs(const OutputVector& new_args
 namespace squeeze
 {
     template <element::Type_t ET>
-    bool evaluate(const HostTensorPtr& arg0, const HostTensorPtr& out)
+    bool evaluate(const HostTensorPtr& arg0, const HostTensorPtr& arg1, const HostTensorPtr& out)
     {
-        runtime::reference::copy(
-            arg0->get_data_ptr<ET>(), out->get_data_ptr<ET>(), shape_size(out->get_shape()));
+        const auto data_rank = arg0->get_partial_shape().rank().get_length();
+        const auto axes_num = shape_size(arg1->get_shape());
+        auto axes = normalize_axes(
+            "",
+            std::vector<int64_t>(arg1->get_data_ptr<ET>(), arg1->get_data_ptr<ET>() + axes_num),
+            data_rank);
+        std::sort(axes.begin(), axes.end(), [](int64_t a, int64_t b) { return a > b; });
+
+        auto out_shape = arg0->get_shape();
+        for (const auto& axis : axes)
+        {
+            out_shape.erase(out_shape.begin() + axis);
```

Comment:
Should we check that specified dimension is equal to 1?

Reply:
Yeah, I think so, done.